velero-2-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,77 @@
+{
+  new(this): {
+
+    groups: [
+      {
+        name: this.config.uid,
+        rules: [
+          {
+            alert: 'VeleroBackupFailure',
+            expr: |||
+              increase(velero_backup_failure_total{%(filteringSelector)s}[5m]) > %(alertsHighBackupFailure)s
+
+            ||| % this.config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Velero backup failures detected.',
+              description: |||
+                Backup failures detected on {{ $labels.instance }}. This could lead to data loss or inability to recover in case of a disaster.
+              ||| % this.config,
+            },
+          },
+          {
+            alert: 'VeleroHighBackupDuration',
+            expr: |||
+              histogram_quantile(0.5, sum(rate(velero_backup_duration_seconds_bucket{%(filteringSelector)s}[5m])) by (le)) > %(alertsHighBackupDuration)s * avg_over_time(histogram_quantile(0.5, sum(rate(velero_backup_duration_seconds_bucket{%(filteringSelector)s}[5m])) by (le))[48h:5m])
```

Comment:
Great alert overall! Have you got any chance to test its behavior? I think it might be prone to produce false positives:
1) aggregating different backup jobs together (by le only). Perhaps, this should be aggregated by backup jobs to have better accuracy?

Reply:
I've set it up and have been keeping an eye on it. I havent seen it produce any false positives yet, but that maybe just luck. I can try doing aggregations by backup jobs as well though. I do not feel strongly about it either way. Let me know what you think or maybe @mshahzeb can weigh in on what he thinks as well

Reply:
If you tend to keep it I suggest two things:
- Split alerts by backup jobs separately so previous day stats are more accurate
- Try add `keep_firing_for: 24h` So alert don't get resolved after 5 minutes after alert is triggered. with keep_firing_for that long I expect it only gets resolved if next day backup is still having issues.